File: 1-js-basics/3.2-conditional-statements-advanced-exercise/06-operationsBetweenNumbers.js
function solve([a, b, operator]) {

    a = Number(a);

    b = Number(b);

    let type = '';

    let data = {

        '+': a + b,

        '-': a - b,

        '*': a * b,

        '/' : a / b,

        '%': a % b

    }

    if (b == 0) {

        return `Cannot divide ${a} by zero`;

    }

    if (operator == '+' || operator == '-' || operator == '*') {

        data[operator] % 2 == 0 ? type = 'even' : type = 'odd';

        return `${a} ${operator} ${b} = ${data[operator]} - ${type}`;

    }

    return operator == '/' ? `${a} ${operator} ${b} = ${data[operator].toFixed(2)}` : `${a} ${operator} ${b} = ${data[operator]}`;

}

console.log(solve(['10', '12', '+']));
console.log(solve(['10', '1', '-']));
console.log(solve(['7', '3', '*']));
console.log(solve(['123', '12', '/']));
console.log(solve(['10', '3', '%']));
console.log(solve(['112', '0', '/']));
console.log(solve(['10', '0', '%']));

File: 2-python/1-python-basics/3.2-conditional-statements-advanced-exercise/06-operations_between_numbers.py
a = int(input())
b = int(input())
operator = input()
result = 0
output = ''

if operator == '+':
    result = a + b
    output = f'{a} + {b} = {result}'
elif operator == '-':
    result = a - b
    output = f'{a} - {b} = {result}'
elif operator == '*':
    result = a * b
    output = f'{a} * {b} = {result}'
else:
    if b == 0:
        output = f'Cannot divide {a} by zero'
    elif operator == '/':
        result = a / b
        output = f'{a} / {b} = {result:.2f}'
    else:
        result = a % b
        output = f'{a} % {b} = {result}'

if operator == '+' or operator == '-' or operator == '*':
    output += ' - even' if result % 2 == 0 else ' - odd'

print(output)
